drivers/include/fxos8700.h

```diff
@@ -0,0 +1,277 @@
+#define FXOS8700_REG_MIN_Z_LSB          (0x50)
+#define FXOS8700_REG_TEMP               (0x51)
+#define FXOS8700_REG_M_THS_CFG          (0x52)
+#define FXOS8700_REG_M_THS_SRC          (0x53)
+#define FXOS8700_REG_M_THS_X_MSB        (0x54)
+#define FXOS8700_REG_M_THS_X_LSB        (0x55)
+#define FXOS8700_REG_M_THS_Y_MSB        (0x56)
+#define FXOS8700_REG_M_THS_Y_LSB        (0x57)
+#define FXOS8700_REG_M_THS_Z_MSB        (0x58)
+#define FXOS8700_REG_M_THS_Z_LSB        (0x59)
+#define FXOS8700_REG_M_THS_COUNT        (0x5A)
+#define FXOS8700_REG_M_CTRL_REG1        (0x5B)
+#define FXOS8700_REG_M_CTRL_REG2        (0x5C)
+#define FXOS8700_REG_M_CTRL_REG3        (0x5D)
+#define FXOS8700_REG_M_INT_SRC          (0x5E)
+#define FXOS8700_REG_A_VECM_CFG         (0x5F)
+#define FXOS8700_REG_A_VECM_THS_MSB     (0x60)
+#define FXOS8700_REG_A_VECM_THS_LSB     (0x61)
+#define FXOS8700_REG_A_VECM_CNT         (0x62)
+#define FXOS8700_REG_A_VECM_INITX_MSB   (0x63)
+#define FXOS8700_REG_A_VECM_INITX_LSB   (0x64)
+#define FXOS8700_REG_A_VECM_INITY_MSB   (0x65)
+#define FXOS8700_REG_A_VECM_INITY_LSB   (0x66)
+#define FXOS8700_REG_A_VECM_INITZ_MSB   (0x67)
+#define FXOS8700_REG_A_VECM_INITZ_LSB   (0x68)
+#define FXOS8700_REG_M_VECM_CFG         (0x69)
+#define FXOS8700_REG_M_VECM_THS_MSB     (0x6A)
+#define FXOS8700_REG_M_VECM_THS_LSB     (0x6B)
+#define FXOS8700_REG_M_VECM_CNT         (0x6C)
+#define FXOS8700_REG_M_VECM_INITX_MSB   (0x6D)
+#define FXOS8700_REG_M_VECM_INITX_LSB   (0x6E)
+#define FXOS8700_REG_M_VECM_INITY_MSB   (0x6F)
+#define FXOS8700_REG_M_VECM_INITY_LSB   (0x70)
+#define FXOS8700_REG_M_VECM_INITZ_MSB   (0x71)
+#define FXOS8700_REG_M_VECM_INITZ_LSB   (0x72)
+#define FXOS8700_REG_A_FFMT_THS_X_MSB   (0x73)
+#define FXOS8700_REG_A_FFMT_THS_X_LSB   (0x74)
+#define FXOS8700_REG_A_FFMT_THS_Y_MSB   (0x75)
+#define FXOS8700_REG_A_FFMT_THS_Y_LSB   (0x76)
+#define FXOS8700_REG_A_FFMT_THS_Z_MSB   (0x77)
+#define FXOS8700_REG_A_FFMT_THS_Z_LSB   (0x78)
+/** @} */
+
+/**
+ * @brief FXOS8700 Device ID
+ */
+#define FXOS8700_WHO_AM_I_VAL           (0xC7)
+
+/**
+ * @brief   FXOS8700 specific return values
+ */
+enum {
+    FXOS8700_OK      = 0,     /**< everything went as expected */
+    FXOS8700_ADDRERR = -1,    /**< no FXOS8700 device found on the bus */
+    FXOS8700_NOBUS   = -2,    /**< errors while initializing the I2C bus */
+    FXOS8700_NODEV   = -3,    /**< no FXOS8700 device found on the bus */
+    FXOS8700_BUSERR  = -4     /**< error during I2C communication */
+};
+
+/**
+ * @brief   Parameters needed for device initialization
+ */
+typedef struct {
+    i2c_t i2c;                /**< I2C device that sensor is connected to */
+    uint8_t addr;             /**< I2C address of this particular sensor */
+    uint32_t renew_interval;  /**< Interval for cache renewal */
+} fxos8700_params_t;
+
+/**
+ * @brief   Device descriptor for a FXOS8700 device
+ */
+typedef struct {
+    fxos8700_measurement_t acc_cached;
```

Comment:
The attributes of this struct are missing documentation, same with the fxos8700_measurement_t struct